include/mbedtls/ecc.h

```diff
@@ -0,0 +1,554 @@
+/* ecc.h - TinyCrypt interface to common ECC functions */
```

Comment:
Note: the location of this file contradicts the design document: "The micro-ecc code should reside in it's own folder to avoid any misunderstandings to what is native to the Mbed TLS and what is not." As a matter of personal preference, I think it's simpler to have it here as it makes the build system simpler, and I think the headers are clear enough regarding what's native to Mbed TLS or not, but I might be biased here. @sbutcher-arm wdyt?

(This comment holds for all 6 uECC files, I won't repeat it for each file.)

Reply:
Hanno was stating for the other option (That the uECC files would be under own dir and because of that comment, the design document is like that), not sure what would be the best option.